install/minikube-installation-guide.md

```diff
@@ -0,0 +1,236 @@
+configured for you) to deploy them:
+
+```sh
+$ git clone https://github.com/kata-containers/packaging.git
+$ cd packaging/kata-deploy
+$ kubectl apply -f kata-rbac.yaml
+$ kubectl apply -f kata-deploy.yaml
+```
+
+This installs the `Kata` components into `/opt/kata` inside the `Minikube` node. It can take
+a few minutes for the operation to complete. Check the installation has worked using:
+
+```sh
+$ minikube ssh
+$ cd /opt/kata
+# If the following directory is not yet created and populated, wait a short while and
+# try again - the download and install process can take a few minutes.
+$ ls bin
+containerd-shim-kata-v2  kata-collect-data.sh  kata-qemu     qemu-ga	     qemu-system-x86_64
+firecracker		 kata-fc	       kata-runtime  qemu-pr-helper  virtfs-proxy-helper
+$ exit
+```
+
+This shows the `Kata` components, including a `qemu`, and the `kata-runtime` for instance.
+
+## Enabling `Kata`
+
+> **Note:** This step is only required for `Minikube`/`k8s` versions <= 1.13. Since version
+> v1.14, the `RuntimeClass` is enabled by default. Performing this step on k8s > v1.14 is
+> however benign.
+
+Now the `Kata` components are installed in the `Minikube` node, you need to configure `k8s`
+`RuntimeClass` to know when to use `Kata` to run a pod.
+
+```sh
+$ curl https://raw.githubusercontent.com/kubernetes/node-api/master/manifests/runtimeclass_crd.yaml > runtimeclass_crd.yaml
+$ kubectl apply -f runtimeclass_crd.yaml
+```
+
+### Register the runtime
+
+Now register the `kata qemu` runtime with that class:
+
+```sh
+$ cd packaging/kata-deploy/k8s-1.14
+$ kubectl apply -f kata-qemu-runtimeClass.yaml
+runtimeclass.node.k8s.io/kata created
+runtimeclass.node.k8s.io/kata-qemu created
+```
+
+`Kata` should now be installed and enabled in the `Minikube` cluster.
+
+## Testing `Kata`
+
+Launch a container that is defined to run on Kata. The enabling is configured by
+the following lines in the yaml file:
+
+```yaml
+    spec:
+      runtimeClassName: kata-qemu
+```
+
+Perform the following action to launch a `Kata` based pod:
+
+```sh
+$ cd packaging/kata-deploy/examples
+$ kubectl apply -f test-deploy-kata-qemu.yaml
+```
+
+An apache php pod should have been deployed with the `Kata` runtime. Wait a few moments
+to check it is running:
+
+```sh
+$ kubectl get pods
+NAME                                   READY   STATUS    RESTARTS   AGE
+php-apache-kata-qemu-bc4c55994-p6f6k   1/1     Running   0          1m
+```
+
+There are a couple of ways to verify it is running with `Kata`. Nominally, it
+should be hard to tell - the idea of `Kata` is that your pod will run inside a VM,
+but look and feel just like it would as a normal software pod. In
+theory, you should not be able to tell your pod is running as a `Kata` container.
+Careful examination can verify your pod is in fact a `Kata` pod.
+
+First, look on the node for a `qemu` running, and which version of the kernel the
+node is running:
+
+```sh
+$ minikube ssh
+$ ps -ef | fgrep qemu-system
```

Comment:
Could this and the `uname` command become two `minikube ssh ...` command instead?